src/orion/algo/evolution_es.py

```diff
@@ -0,0 +1,272 @@
+Params: {params}
+"""
+
+SPACE_ERROR = """
+EvolutionES cannot be used if space does not contain a fidelity dimension.
+"""
+
+BUDGET_ERROR = """
+Cannot build budgets below max_resources;
+(max: {}) - (min: {}) > (num_rungs: {})
+"""
+
+
+def compute_budgets(min_resources, max_resources, reduction_factor, nums_population, pairs):
+    """Compute the budgets used for each execution of hyperband"""
+    budgets_eves = []
+    if reduction_factor == 1:
+        for i in range(min_resources, max_resources + 1):
+            if i == min_resources:
+                budgets_eves.append([(nums_population, i)])
+            else:
+                budgets_eves[0].append((pairs * 2, i))
+    else:
+        num_brackets = int(np.log(max_resources) / np.log(reduction_factor))
+        budgets = []
+        budgets_tab = {}  # just for display consideration
+        for bracket_id in range(0, num_brackets + 1):
+            bracket_budgets = []
+            num_trials = int(np.ceil(int((num_brackets + 1) / (num_brackets - bracket_id + 1)) *
+                                     (reduction_factor ** (num_brackets - bracket_id))))
+
+            min_resources = max_resources / reduction_factor ** (num_brackets - bracket_id)
+            for i in range(0, num_brackets - bracket_id + 1):
+                n_i = int(num_trials / reduction_factor ** i)
+                min_i = int(min_resources * reduction_factor ** i)
+                bracket_budgets.append((n_i, min_i))
+
+                if budgets_tab.get(i):
+                    budgets_tab[i].append((n_i, min_i))
+                else:
+                    budgets_tab[i] = [(n_i, min_i)]
+
+            budgets.append(bracket_budgets)
+
+        for i in range(len(budgets[0])):
+            if i == 0:
+                budgets_eves.append([(nums_population, budgets[0][i][1])])
+            else:
+                budgets_eves[0].append((pairs * 2, budgets[0][i][1]))
+
+    return budgets_eves
+
+
+class EvolutionES(Hyperband):
+    """EvolutionES formulates hyperparameter optimization as an evolution.
+
+    For more information on the algorithm,
+    see original paper at
+    https://arxiv.org/pdf/1703.01041.pdf and
+    https://arxiv.org/pdf/1901.11117.pdf
+
+    Real et al. "Large-Scale Evolution of Image Classifiers"
+    So et all. "The Evolved Transformer"
+
+    Parameters
+    ----------
+    space: `orion.algo.space.Space`
+        Optimisation space with priors for each dimension.
+    seed: None, int or sequence of int
+        Seed for the random number generator used to sample new trials.
+        Default: ``None``
+    repetitions: int
+        Number of execution of Hyperband. Default is numpy.inf which means to
+        run Hyperband until no new trials can be suggested.
+
+    """
+
+    def __init__(self, space, seed=None, repetitions=np.inf, nums_population=20, mutate=None):
+        super(EvolutionES, self).__init__(space, seed=seed, repetitions=repetitions)
```

Comment:
I think we should verify that nums_population is even otherwise we will have unbalanced teams.

Reply:
I considered this case. In the code:  pair = nums_population // 2, which ensure the teams are balanced. User doesn't need to care about the nums_population is even or odd, if it is odd, for example 9, then the blue team and red team are both 4 individuals. It is transparent to the user.